src/trackerHook.js

```diff
@@ -36,43 +51,39 @@ export const usePromiseTracker = (outerConfig = defaultConfig) => {
     internalPromiseInProgress
   );
 
-  const notifiyPromiseInProgress = () => {
-    (!config || !config.delay || config.delay === 0) ?
-      setPromiseInProgress(true)
-    :
-      setTimeout(() => {
-        // Check here ref to internalPromiseInProgress
-        if (latestInternalPromiseInProgress.current) {
-          setPromiseInProgress(true);
-        }
-      }, config.delay);
+  const notifyPromiseInProgress = () => {
+    !config || !config.delay || config.delay === 0
+      ? setPromiseInProgress(true)
+      : setTimeout(() => {
+          // Check here ref to internalPromiseInProgress
+          if (latestInternalPromiseInProgress.current) {
+            setPromiseInProgress(true);
+          }
+        }, config.delay);
   };
 
   const updatePromiseTrackerStatus = (anyPromiseInProgress, areaAffected) => {
-    if (config.area === areaAffected) {
+    if (isMounted.current && config.area === areaAffected) {
       setInternalPromiseInProgress(anyPromiseInProgress);
       // Update the ref object as well, we will check it when we need to
       // cover the _delay_ case (setTimeout)
       latestInternalPromiseInProgress.current = anyPromiseInProgress;
       if (!anyPromiseInProgress) {
         setPromiseInProgress(false);
       } else {
-        notifiyPromiseInProgress();
+        notifyPromiseInProgress();
       }
     }
   };
 
+  const fn = (anyPromiseInProgress, areaAffected) =>

```

Comment:
In think this fn is not necessary, could you try ` emitter.on(promiseCounterUpdateEventId, updatePromiseTrackerStatus);` and `return () => emitter.off(promiseCounterUpdateEventId, updatePromiseTrackerStatus);` , please?

Reply:
Done.